public/src/managers/ServerProvider.ts

```diff
@@ -243,6 +243,67 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
         return JSON.parse(JSON.stringify(result.statusCode)) as IStatusCode;
     }
 
+    // getAllGroupLabInfos 
+    public async getAllGroupLabInfos(courseId: number, groupID: number): Promise<IMap<ISubmission>> { 
+        const result = await this.helper.get<ISubmission[]>(
+            ("courses/" + courseId.toString() + "/groups/" + groupID + "/submissions"),
```

Comment:
Could this type of string concatenation be replaced with [template strings](https://basarat.gitbooks.io/typescript/docs/template-strings.html).

Alternatively, we could use an `Array` and `join("/")` as shown [here](https://www.tutorialspoint.com/typescript/typescript_array_join.htm). But this approach may not be appropriate in this particular case, unless you move the concatenation before the `get<>` call. 

Another alternative that may be the best, would be create a URL joiner function that takes a variadic list of strings and does a `join("/")`. This function could then be reused everywhere that we want to concat a URL.

Another concern I have with these elements being strings is that it is quite easy to misspell one of these path elements, and so, perhaps it would be useful to create a set of constants that we import everywhere, and use as replacement for strings.

Reply:
I think it could be replaced with template strings, but we still need to differentiate the different placeholders.  Maybe end up with something like

```
public async getAllGroupLabInfos(courseId: number, groupID: number): Promise<IMap<ISubmission>> { 
    var  url = concatenate "courses/${courseID}/groups/${groupID}/submissions";
           const result = await this.helper.get<ISubmission[]>(
            (url),);
}
```

I like the idea of having a set of constants for the different endpoints on the backend. Using them with the string templates maybe?
Where should these constants be defined? Inside an own file for a better overview or inside the ServerProvider? Maybe they should be defined inside ServerProvider as this component can be changed out if the network layer changes.

Reply:
Then if a endpoint name changes, we only have to rename it once, instead of looking through many functions to rename them, and possibly miss some of them.

Reply:
LGTM. Although I would keep each endpoint in the "map" should also be given its own line.

Feel free to commit and push changes to the branch, and I'll review and do edits or suggest changes.